__init__.py

```diff
@@ -15,7 +15,44 @@ class MarkExperience(Event):
 
 
 MODIFIER_REGEX = "[+-]?[0,1,2,3]"
-STAT_REGEX = f"(?:(?:!cool|!tough|!sharp|!charm|!weird) {MODIFIER_REGEX})"
+GAME_STATS = {"motw": ["cool", "tough", "sharp", "charm", "weird"],
+              "pbtastartrek": ["aggressive", "bold", "talk", "tech"]}
+stat_regexes = {}
+
+@match_regex(r"\!set game ?(?P<gamename>.*)", case_sensitive=False)
+@memory_in_event_room
+async def set_game(opsdroid, config, message):
+    game = message.regex.capturesdict()['gamename'][0]
+    if game not in GAME_STATS.keys():
+        message.respond(f"I don't know how to play that. Available options are: {', '.join(GAME_STATS.keys())}")
+        return
+    await opsdroid.memory.put(f"{game}_stats", GAME_STATS[game])
+
+
+async def get_game(room):
+    with db.memory_in_room(room):
+        game = await opsdroid.memory.get("game_rules") or {}
+    return game
+
+
+async def get_stats(room):
+    game = await get_game(room)
+    stats = GAME_STATS[game] if game else []
+    return stats
+
+
+def html_list(sequence):
+    html_stats = ''.join([f"<li>{s.capitalize()}</li>" for s in sequence])
+    return f"<ul>{html_stats}</ul>"
+
+
+async def filter_by_game_stats(string, room, db):
+    if room not in stat_regexes.keys():
+        gamestats = await get_stats(room)
+        stats_re = f"(?:(?:{'|'.join(['!'+s for s in gamestats])}) {MODIFIER_REGEX})"
```

Comment:
I was thinking you could actually put the compiled regex object in this cache, make it marginally faster.

Reply:
You mean the return value from `.findall()`? Surely that will vary depending on the input string?

Reply:
I was thinking this:
```suggestion
        stats_re = regex.compile(f"(?:(?:{'|'.join(['!'+s for s in gamestats])}) {MODIFIER_REGEX})")
```
https://docs.python.org/3/library/re.html#re.compile